### PR TITLE
Update ORT training examples & add summarization example

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

- [x] Replace metrics in the example from datasets to evaluate.
- [x] Add summarization example for `ORTSeq2SeqTrainer`.
- [x] Enable `inference_with_ort` in training examples.
- [x] Update Readme to introduce inference with ORT.
- [x] Update examples to to be compatible with transformers 4.22.1

### CODE REVIEW

**Comment by JingyaHuang** [comment]
P.S. This PR is based on the implementation of ORT inference for `ORTSeq2SeqTrainer`,  it is built on the top of PR #189 .

File: optimum/onnxruntime/trainer.py [hunk 1]
```diff
@@ -740,28 +734,28 @@ def predict(
         start_time = time.time()
 
         if inference_with_ort:
+            logger.warning("[INFO] Predicting with ONNX Runtime backend.")
```
Comment (regisss) [inline optimum/onnxruntime/trainer.py:737]: Same

File: optimum/onnxruntime/trainer.py [hunk 2]
```diff
@@ -740,28 +734,28 @@ def predict(
...
-                )
-            except Exception as error:
-                logger.error(error)
-                logger.warning(
-                    f"[ERROR!] Prediction with ONNX Runtime is not available with {self.model.config.name_or_path} model. Remove `inference_with_ort` to predict within PyTorch."
-                )
-                raise
         else:
             logger.warning(
                 "[INFO] Predicting with PyTorch backend. If you want to use ONNX Runtime for the prediction, set `trainer.predict(inference_with_ort=True)`."
```
Comment (regisss) [inline optimum/onnxruntime/trainer.py:741]: Same

File: optimum/onnxruntime/trainer.py [hunk 3]
```diff
@@ -740,28 +734,28 @@ def predict(
...
+                test_dataloader,
+                description="Prediction",
+                ignore_keys=ignore_keys,
+                metric_key_prefix=metric_key_prefix,
             )
+        except Exception as error:
+            logger.error(error)
+            if inference_with_ort:
+                logger.warning(
+                    f"[ERROR!] Prediction with ONNX Runtime is not available for {self.model.config.name_or_path} model. Set `inference_with_ort=False` to predict with PyTorch."
```
Comment (regisss) [inline optimum/onnxruntime/trainer.py:756]: Same

File: examples/onnxruntime/training/summarization/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Summarization
+
+By running the script [`run_summarization.py`](https://github.com/huggingface/optimum/blob/main/examples/onnxruntime/training/summarization/run_summarization.py),
+you will be able to leverage the [`ONNX Runtime`](https://github.com/microsoft/onnxruntime) to fine-tune and evaluate models from the
```
Comment (regisss) [inline examples/onnxruntime/training/summarization/README.md:20]:
```suggestion
you will be able to leverage the [`ONNX Runtime`](https://github.com/microsoft/onnxruntime) accelerator to fine-tune and evaluate models from the
```

File: examples/onnxruntime/training/summarization/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+[HuggingFace hub](https://huggingface.co/models) on summarization tasks.
+
+### Supported models
```
Comment (regisss) [inline examples/onnxruntime/training/summarization/README.md:23]: `##` no? Or you prefer the rendering of `###`?

File: examples/onnxruntime/training/summarization/README.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+* `Bart`
+* `T5`
+
+`run_summarization.py` is a lightweight example of how to download and preprocess a dataset from the 🤗 Datasets library or use your own files (jsonlines or csv), then fine-tune one of the architectures above on it.
+
+
+__The following example applies the acceleration features powered by ONNX Runtime.__
+
+
+### Onnxruntime Training
```
Comment (regisss) [inline examples/onnxruntime/training/summarization/README.md:36]:
```suggestion
### Onnx Runtime Training
```
And same comment for `##`.

File: examples/onnxruntime/training/summarization/README.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+# Summarization
+
+By running the script [`run_summarization.py`](https://github.com/huggingface/optimum/blob/main/examples/onnxruntime/training/summarization/run_summarization.py),
+you will be able to leverage the [`ONNX Runtime`](https://github.com/microsoft/onnxruntime) to fine-tune and evaluate models from the
+[HuggingFace hub](https://huggingface.co/models) on summarization tasks.
+
+### Supported models
+
+Theorectically, all sequence-to-sequence models with [ONNXConfig](https://github.com/huggingface/transformers/blob/main/src/transformers/onnx/features.py) support in Transformers shall work, here are models that Optimum team has tested and validated.
```
Comment (regisss) [inline examples/onnxruntime/training/summarization/README.md:25]:
```suggestion
Theoretically, all sequence-to-sequence models with [ONNXConfig](https://github.com/huggingface/transformers/blob/main/src/transformers/onnx/features.py) support in Transformers shall work, here are the models that the Optimum team has tested and validated.
```

File: examples/onnxruntime/training/summarization/README.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+python run_summarization.py \
+    --model_name_or_path t5-small \
+    --dataset_name cnn_dailymail \
+    --dataset_config "3.0.0" \
+    --source_prefix "summarize: " \
+    --do_train \
+    --do_eval \
+    --per_device_train_batch_size=4 \
+    --per_device_eval_batch_size=4 \
+    --output_dir /tmp/ort_summarization_t5/
```
Comment (regisss) [inline examples/onnxruntime/training/summarization/README.md:50]:
```suggestion
    --output_dir /tmp/ort_summarization_t5/ \
```

**Review by regisss — COMMENTED** [review]
Mostly typos.
I just have a question regarding the use of `[INFO]` or `[ERROR]` with `logger.warning`. It looks weird to me but I guess there is a good reason to do it :slightly_smiling_face:

File: optimum/onnxruntime/trainer.py [hunk 9]
```diff
@@ -672,28 +672,15 @@ def evaluate(
         start_time = time.time()
 
         if inference_with_ort:
+            logger.warning("[INFO] Evaluating with ONNX Runtime backend.")
```
[the remark this answers is not in the document]
Comment (JingyaHuang) [reply optimum/onnxruntime/trainer.py:678]: @regisss You are right, the loggers are not all consistent, I will put them as it is supposed to. I think that I once put them this way as at that time the inference with ort was experimental.

File: examples/onnxruntime/training/summarization/README.md (hunk 5 above)
Comment (JingyaHuang) [reply examples/onnxruntime/training/summarization/README.md:23]: transformers put the section as ###, I will probably keep it that way.

**Review by regisss — APPROVED** [review]
LGTM @JingyaHuang !!

**Pull request merged**